### PR TITLE
Fixed #34692  -- Made autoreloader reset cached template loader for default renderer.

### PR DESCRIPTION
**Pull request by priyank-panchal**
https://code.djangoproject.com/ticket/34692

### CODE REVIEW

**Comment by smithdc1** [comment]
Thanks for the patch. 👍 

Please could a test for this regression be added. 

Also, I see this just targets the DjangoTemplates backend.  What do you about the others, for example Jinja2. Are these also impacted?

**Comment by priyank-panchal** [comment]
Yes, it also works with Jinja2. I have tested it.

**Comment by smithdc1** [comment]
@priyank-panchal super. If you could add a test case for that as well, that would be super. 🎁

**Comment by priyank-panchal** [comment]
@smithdc1, sorry for the late reply. I have added the test case. If you have any suggestions, I'd be happy to consider them.

File: tests/template_tests/test_autoreloader.py [hunk 1]
```diff
@@ -56,6 +56,17 @@ def test_non_template_changed_in_template_directory(self, mock_reset):
         self.assertIsNone(autoreload.template_changed(None, Path(__file__)))
         mock_reset.assert_not_called()
 
+    @mock.patch("django.forms.renderers.get_default_renderer")
+    def test_form_template_reset(self, mock_reset):
+        template_path = Path(__file__).parent / "templates" / "index.html"
+        self.assertTrue(autoreload.template_changed(None, template_path))
+        mock_reset.assert_called_once()
```
Comment (felixxm) [inline tests/template_tests/test_autoreloader.py:63]: We should check that `reset()` is called.

File: tests/template_tests/test_autoreloader.py [hunk 2]
```diff
@@ -56,6 +56,17 @@ def test_non_template_changed_in_template_directory(self, mock_reset):
         self.assertIsNone(autoreload.template_changed(None, Path(__file__)))
         mock_reset.assert_not_called()
 
+    @mock.patch("django.forms.renderers.get_default_renderer")
+    def test_form_template_reset(self, mock_reset):
```
Comment (felixxm) [inline tests/template_tests/test_autoreloader.py:60]:
```suggestion
    def test_form_template_reset_template_change(self, mock_reset):
```

File: tests/template_tests/test_autoreloader.py [hunk 3]
```diff
@@ -56,6 +56,17 @@ def test_non_template_changed_in_template_directory(self, mock_reset):
...
+        template_path = Path(__file__).parent / "templates" / "index.html"
+        self.assertTrue(autoreload.template_changed(None, template_path))
+        mock_reset.assert_called_once()
+
+    @mock.patch("django.forms.renderers.get_default_renderer")
+    def test_form_template_not_reset(self, mock_reset):
```
Comment (felixxm) [inline tests/template_tests/test_autoreloader.py:66]:
```suggestion
    def  test_form_template_reset_non_template_change(self, mock_reset):
```

File: tests/template_tests/test_autoreloader.py [hunk 4]
```diff
@@ -56,6 +56,17 @@ def test_non_template_changed_in_template_directory(self, mock_reset):
         self.assertIsNone(autoreload.template_changed(None, Path(__file__)))
         mock_reset.assert_not_called()
 
+    @mock.patch("django.forms.renderers.get_default_renderer")
+    def test_form_template_reset(self, mock_reset):
+        template_path = Path(__file__).parent / "templates" / "index.html"
+        self.assertTrue(autoreload.template_changed(None, template_path))
```
Comment (felixxm) [inline tests/template_tests/test_autoreloader.py:62]:
```suggestion
        self.assertIs(autoreload.template_changed(None, template_path), True)
```
This guideline is from [Python coding style](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/coding-style/#python-style).

**Review by felixxm — COMMENTED** [review]
@priyank-panchal Thanks :+1: I left comments.

**Comment by felixxm** [comment]
@priyank-panchal Do you have time to keep working on this?

**Pull request merged**